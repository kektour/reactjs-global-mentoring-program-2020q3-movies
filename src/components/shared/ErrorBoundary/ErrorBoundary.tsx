import React from 'react';
import styles from './ErrorBoundary.module.scss';

type Props = {};
type State = {
  hasError: boolean;
}

export class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError() {
    return { hasError: true };
  }

  render() {
    const { hasError } = this.state;
    if (hasError) {
      return (
        <div className={styles.root}>
          <h1 className={styles.errText}>Error. Something went wrong</h1>;
        </div>
      );
    }

    return this.props.children;
  }
}
