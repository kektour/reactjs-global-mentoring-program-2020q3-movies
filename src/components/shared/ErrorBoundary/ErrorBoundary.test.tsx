import React from 'react';
import { render } from '@testing-library/react';
import { ErrorBoundary } from './ErrorBoundary';

describe('ErrorBoundary', () => {
  it('Should render component correctly', () => {
    const { asFragment } = render(
      <ErrorBoundary>
        <div>Foo</div>
      </ErrorBoundary>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render error message', () => {
    const StubComponent: React.FC<{}> = () => {
      throw new Error('Foo');
    };
    const { asFragment } = render(
      <ErrorBoundary>
        <StubComponent />
      </ErrorBoundary>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
