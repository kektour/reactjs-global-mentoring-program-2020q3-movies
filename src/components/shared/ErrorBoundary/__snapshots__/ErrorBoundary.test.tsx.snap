// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Should render component correctly 1`] = `
<DocumentFragment>
  <div>
    Foo
  </div>
</DocumentFragment>
`;

exports[`ErrorBoundary Should render error message 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <h1
      class="errText"
    >
      Error. Something went wrong
    </h1>
    ;
  </div>
</DocumentFragment>
`;
